Binning tutorial update (#6486)
galaxyproject/training-material: topics/microbiome/tutorials/metagenomics-binning/tutorial.md

Reviewer: ```suggestion
```

  of research areas, such as human health, environmental microbiology, and biotechnology
contributions:
  authorship:
  - paulzierep
  - npechl
  - fpsom

>   - vinisalazar

Reviewer: ```suggestion
  - vinisalazar
  editing:
  - bebatut
```

---

Metagenomics is the study of genetic material recovered directly from environmental samples, such as soil, water, or gut contents, without the need for isolation or cultivation of individual organisms. Metagenomics binning is a process used to classify DNA sequences obtained from metagenomic sequencing into discrete groups, or bins, based on their similarity to each other.

> The goal of metagenomics binning is to assign the DNA sequences to the organisms or taxonomic groups that they originate from, allowing for a better understanding of the diversity and functions of the microbial communities present in the sample. This is typically achieved through computational methods that include sequence similarity, composition, and other features to group the sequences into bins.

Reviewer: ```suggestion
The goal of metagenomics binning is to assign DNA sequences to the organisms or taxonomic groups from which they originate, allowing for a better understanding of the diversity and functions of the microbial communities present in the sample. This is typically achieved through computational methods that include sequence similarity, composition, and other features to group the sequences into bins.
```

- **Chimeric sequences**: Sequences that are the result of sequencing errors or contamination can lead to chimeric sequences, which can make it difficult to accurately bin reads.
- **Strain variation**: Organisms within a species can exhibit significant genetic variation, which can make it difficult to distinguish between different strains in a metagenomic sample.

## Common binners

There are plenty of algorithms that perform metagenomic binning. Some of the most widely used include:

> * **MaxBin** ({%cite maxbin2015%}): A popular de novo binning algorithm that uses a combination of sequence features and marker genes to cluster contigs into genome bins.

Reviewer: ```suggestion
* **MaxBin** ({% cite maxbin2015 %}): A popular de novo binning algorithm that uses a combination of sequence features and marker genes to cluster contigs into genome bins.
```

* **MetaBAT** ({%cite Kang2019%}): Another widely used de novo binning algorithm that employs a hierarchical clustering approach based on tetranucleotide frequency and coverage information.
* **CONCOCT** ({%cite Alneberg2014%}): A de novo binning tool that uses a clustering algorithm based on sequence composition and coverage information to group contigs into genome bins.
* **MyCC** ({%cite Lin2016%}): A reference-based binning tool that uses sequence alignment to identify contigs belonging to the same genome or taxonomic group.

> * **GroopM** ({%cite Imelfort2014%}): A hybrid binning tool that combines reference-based and de novo approaches to achieve high binning accuracy.

Reviewer: ```suggestion
* **GroopM** ({% cite Imelfort2014 %}): A hybrid binning tool that combines reference-based and de novo approaches to achieve high binning accuracy.
```

* **SemiBin** ({%cite Pan2022%}): A command-line tool for metagenomic binning with deep learning; handles both short and long reads.

> * **Vamb** ({%cite nissen2021improved%}): An algorithm that uses variational autoencoders (VAEs) to encode sequence composition and coverage information.

Reviewer: ```suggestion
* **Vamb** ({% cite nissen2021improved %}): An algorithm that uses variational autoencoders (VAEs) to encode sequence composition and coverage information.
```

# Prepare analysis history and data

Metagenomic binners take typically two data typs as input: assembled contigs in fasta format and coverage information in bam format.

- A fasta file containing the assembled contigs, which can be generated from raw metagenomic sequencing reads using an assembler such as MEGAHIT, SPAdes, or IDBA-UD.

> - A bam file containing the read coverage information for each contig, which can be generated from the same sequencing reads using mapping software such as Bowtie2 or BWA.

Reviewer: ```suggestion
- A BAM file containing the read coverage information for each contig, which can be generated from the same sequencing reads using mapping software such as Bowtie2 or BWA.
```

> <comment-title>Can Bins be generated without coverage information</comment-title>
>
> Not all binners require coverage information — some, like MetaBAT2, can operate using only genomic composition (e.g. tetranucleotide frequencies) when coverage files are not available. This is especially useful for single-sample datasets or legacy data where coverage cannot easily be calculated.
>
> Other tools that support composition-only binning include:

> > - **MaxBin 2** (can run with composition alone, but performs better with depth)

Reviewer: ```suggestion
> - **MaxBin 2**, which can run with composition alone, but performs better with depth.
```

> > - **SolidBin** (supports single-sample binning based on sequence features)

Reviewer: ```suggestion
> - **SolidBin**, which supports single-sample binning based on sequence features.
```

> > - **VAMB** (primarily uses deep learning on composition, coverage optional)

Reviewer: ```suggestion
> - **VAMB**, which primarily uses deep learning on composition, coverage optional.
```

>    > In case of large dataset, we can use FTP server or the [Galaxy Rule-based Uploader]({% link topics/galaxy-interface/tutorials/upload-rules/tutorial.md %}).

> >    {: .comment}

Reviewer: ```suggestion

```

>
> 2. Create a collection named `Contigs`
>
>    {% snippet faqs/galaxy/collections_build_list.md %}
>

> > 3. Also import the raw reads in fastq format (\*.fasta) from [Zenodo]({{ page.zenodo_link }}) or a data library:

Reviewer: ```suggestion
> 3. Import the raw reads in fastq format from [Zenodo]({{ page.zenodo_link }}) or a data library:
```

Make sure the reads are quality controlled. E.g. following the QC toturial TODO.

> <hands-on-title> Map reads to contigs </hands-on-title>
>
> 1. {% tool [Bowtie2](toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.4+galaxy0) %} with the following parameters:
>    - *"Is this single or paired library"*: `Paired-end`
>        - {% icon param-collection %} *"FASTQ Paired Dataset"*: `Reads` (Input dataset collection)
>        - *"Do you want to set paired-end options?"*: `No`
>    - *"Will you select a reference genome from your history or use a built-in index?"*: `Use a genome from the history and build index`
>        - {% icon param-collection %} *"Select reference genome"*: `Contigs` (Input dataset collection)
>    - *"Set read groups information?"*: `Do not set`
>    - *"Select analysis mode"*: `1: Default setting only`
>    - *"Do you want to tweak SAM/BAM Options?"*: `No`
>    - *"Save the bowtie2 mapping statistics to the history"*: `Yes`
>
{: .hands_on}

> > <hands-on-title> Sort bam files </hands-on-title>

Reviewer: ```suggestion
Let's now sort the BAM files.

> <hands-on-title> Sort BAM files </hands-on-title>
```

>
> 1. {% tool [Samtools sort](toolshed.g2.bx.psu.edu/repos/devteam/samtools_sort/samtools_sort/2.0.7) %} with the following parameters:
>    - {% icon param-file %} *"BAM File"*: output of **Bowtie2** {% icon tool %}
>    - *"Primary sort key"*: `coordinate`
>
{: .hands_on}

> The sorted bam file can be used as input for any of the binning tools.

Reviewer: ```suggestion
The sorted BAM file can be used as input for any of the binning tools.
```

In this tutorial, we offer dedicated versions, which highlight each of the following binners:

{% include _includes/cyoa-choices.html option1="MetaBAT2" option2="MaxBin2" option3="SemiBin" option4="CONCOCT" option5="COMEBin" default="MetaBAT2" %}

<div class="MetaBAT2" markdown="1">
{% include topics/microbiome/tutorials/metagenomics-binning/metabet2_version.md %}
</div>
<div class="MaxBin2" markdown="1">
{% include topics/microbiome/tutorials/metagenomics-binning/maxbin2_version.md %}
</div>
<div class="SemiBin" markdown="1">
{% include topics/microbiome/tutorials/metagenomics-binning/semibin_version.md %}
</div>
<div class="CONCOCT" markdown="1">
{% include topics/microbiome/tutorials/metagenomics-binning/concoct_version.md %}
</div>
<div class="COMEBin" markdown="1">
{% include topics/microbiome/tutorials/metagenomics-binning/comebin_version.md %}
</div>

# Bin refinement

Now, that you have produced bins with your favorite Binning algorithms you can refine the recovered bins. 
Therefore, you need to convert the bins into a contig to bin mapping table, combine the tables from each binner into one collection and

> use Binette to creat consensus bins. An alternative tool would be {% tool [DAS Tool](toolshed.g2.bx.psu.edu/repos/iuc/das_tool/das_tool/1.1.7+galaxy1) %} which is also available in Galaxy.

Reviewer: ```suggestion
Therefore, we need to 

1. Convert the bins into a contig-to-bin mapping table,
2. Combine the tables from each binner into one collection, and
3. Use **Binette** ({% cite Mainguy2024Binette %}) to create consensus bins.

   An alternative tool would be {% tool [DAS Tool](toolshed.g2.bx.psu.edu/repos/iuc/das_tool/das_tool/1.1.7+galaxy1) %} which is also available in Galaxy.
```

`CheckM lineage_wf` is a specific workflow within the CheckM software tool that is used for taxonomic classification of genome bins based on their marker gene content. This workflow uses a reference database of marker genes and taxonomic information to classify the genome bins at different taxonomic levels, from domain to species.

> Now you can investigate the completeness and contamination of any of your previously generated genome bins as well as the refined set.

Reviewer: ```suggestion
Now we can investigate the completeness and contamination of any of your previously generated genome bins, as well as the refined set.
```

> > | 8      | root (UID1)             | 5656      | 56         | 24             | 56 | 0  | 0  | 0  | 0  | 0  | 0.00         | 0.00          | 0.00                  |
> > | 9      | k__Bacteria (UID203)    | 5449      | 103        | 58             | 89 | 14 | 0  | 0  | 0  | 0  | 15.67        | 0.00          | 0.00                  |
> >
> > **Binette**
> >
> > | Bin ID        | Marker lineage        | # genomes | # markers | # marker sets | 0  | 1  | 2 | 3 | 4 | 5+ | Completeness | Contamination | Strain heterogeneity |
> > |---------------|------------------------|-----------|------------|----------------|----|----|----|----|----|----|--------------|---------------|-----------------------|
> > | binette_bin1  | k__Bacteria (UID203)   | 5449      | 103        | 58             | 89 | 14 | 0  | 0  | 0  | 0  | 15.67        | 0.00          | 0.00                  |
> > | binette_bin2  | root (UID1)            | 5656      | 56         | 24             | 56 | 0  | 0  | 0  | 0  | 0  | 0.00         | 0.00          | 0.00                  |
> >

> > > **MetaBAT2** and **SemiBin** generated the best‐performing bins in this dataset, each recovering one bin with ~15.7% completeness and no detectable contamination. 

Reviewer: ```suggestion
```

> >

> > > **MaxBin** produced bins of lower completeness (10.5% and 6.9%), but still without contamination. 

Reviewer: ```suggestion
```

> > >

Reviewer: ```suggestion
```

> > > **CONCOCT** largely failed to recover meaningful bins in this dataset: most bins show 0% completeness. This often occurs with CONCOCT on small assemblies or uneven coverage, because its Gaussian clustering model struggles when the number of contigs is low or the variation in coverage is insufficient.

Reviewer: ```suggestion
```

> >
> > **Binette** produced one bin essentially identical in quality to the MetaBAT2/SemiBin bins (~15.7% completeness), plus a root-level bin with no completeness. Binette therefore could recover the most completely bin sucessful. The additional

> loq quality bin, would have normally be filtered by Binette if the `Set minimus completeness` parameter would be set to a reasonable values, often > 75 % for true biological data.

Reviewer: ```suggestion
loq quality bin, would have normally been filtered by Binette if the `Set minimus completeness` parameter were set to a reasonable value, often > 75 % for true biological data.
```

> {: .solution}
>
{: .question}

> > <comment-title>CheckM2</comment-title>

Reviewer: Maybe better as a detail box. What do you think?

> > CheckM2 ({%cite Chklovski2023CheckM2%}) is the successor of CheckM, but CheckM is still widely used, since its marker-based logic can be more interpretable in a biological sense. E.g. to date (2025-11-21) NCBI still allows to submit MAGs to genbank if either checkM or checkM2 has a completeness of > 90% (see the [NCBI WGS/MAG submission guidelines](https://www.ncbi.nlm.nih.gov/genbank/wgsfaq/#metagen)).

Reviewer: ```suggestion
> CheckM2 ({%cite Chklovski2023CheckM2%}) is the successor of CheckM, but CheckM is still widely used, since its marker-based logic can be more interpretable in a biological sense. E.g., to date (2025-11-21), NCBI still allows submitting MAGs to GenBank if either checkM or checkM2 has a completeness of > 90% (see the [NCBI WGS/MAG submission guidelines](https://www.ncbi.nlm.nih.gov/genbank/wgsfaq/#metagen)).
```

Several tools have been designed for the proccess of de-replication. **`dRep`** is a software tool designed for the dereplication of genomes in metagenomic datasets. The goal is to retain a representative set of genomes to improve downstream analyses, such as taxonomic profiling and functional annotation.

An typical workflow of how `dRep` works for dereplication in metagenomics includes:

- *Genome Comparison*: `dRep` uses a pairwise genome comparison approach to assess the similarity between genomes in a given metagenomic dataset.

- *Clustering*: Based on the genome similarities, `dRep` performs clustering to group similar genomes into "genome clusters." Each cluster represents a group of closely related genomes.

- *Genome Quality Assessment*: `dRep` evaluates the quality of each genome within a cluster. It considers factors such as completeness, contamination, and strain heterogeneity.

> - *Genome Selection*: Within each genome cluster, `dRep` selects a representative genome based on user-defined criteria. This representative genome is considered as the "dereplicated" version of the cluster.

Reviewer: ```suggestion
- *Genome Selection*: Within each genome cluster, dRep selects a representative genome based on user-defined criteria. This representative genome is considered the "dereplicated" version of the cluster.
```

> 4. Bin each assembly separately using your favorite binner
> 5. Bin co-assembly using your favorite binner
> 6. Pull the bins from all assemblies together

> > 7. Run **`dRep`** on them

Reviewer: ```suggestion
> 7. Run {% tool [dRep dereplication](toolshed.g2.bx.psu.edu/repos/iuc/drep_dereplicate/drep_dereplicate/3.6.2+galaxy1) %} on them
```
